### ISSUE TITLE
Lint rsaEncryption algorithm identifiers for mandatory NULL params

### ISSUE THREAD

**Issue by cpu** [zmap/zlint#283]
[RFC 3279, Section 2.3.1 "RSA Keys"](https://tools.ietf.org/html/rfc3279#section-2.3.1) says the following about the `rsaEncryption` algorithm identifier encoding:

```
   The rsaEncryption OID is intended to be used in the algorithm field
   of a value of type AlgorithmIdentifier.  The parameters field MUST
   have ASN.1 type NULL for this algorithm identifier.
```

It was noted by [@sleevi on MDSP](https://groups.google.com/d/msg/mozilla.dev.security.policy/NFAG2_o3dhY/gQ2UW_ohAwAJ) that there are real-world certificates that violate this aspect of RFC 3279 and are flagged by `certlint` but not `zlint`.

Zlint should have a lint that ensures the required `NULL` parameters field of `rsaEncryption` algorithm identifiers in certificates is present. 

Here's an [example ASN.1 decoding](https://lapo.it/asn1js/#MIIFDDCCA_SgAwIBAgIMLdcgkQWz0AYwAS3CMA0GCSqGSIb3DQEBCwUAMEwxCzAJBgNVBAYTAkJFMRkwFwYDVQQKExBHbG9iYWxTaWduIG52LXNhMSIwIAYDVQQDExlBbHBoYVNTTCBDQSAtIFNIQTI1NiAtIEcyMB4XDTE4MTIxMDExMDczM1oXDTIwMDEyOTEzMzQxNVowTzELMAkGA1UEBhMCTkwxITAfBgNVBAsTGERvbWFpbiBDb250cm9sIFZhbGlkYXRlZDEdMBsGA1UEAxMUd3d3LnNob3JlYXJjaGllZi5jb20wggEgMAsGCSqGSIb3DQEBAQOCAQ8AMIIBCgKCAQEAtdAllnmhxoule45cByFJjIuYiRe5YllK0RWYZ8Q9wehjlYTCde5fOtcMigtrYCUheB7Mf173m-xrp8II23WfNK7CIKlQUwuN-blPp-lvxhEQF5p-KaAt0OPnm5WKc67Dwxy6rwztN4O6-mBJpluKYFNHET8auks822aSFH3bvagLebUWxTINxl6RfeM0W7bfZDC8i-DaRfxGrgjgz3pko7og_qy4Dp0rMvFWYSPCT_MspXR3KOrWLrqY-Woa8e2ZDz5TymEYzQPF7hJa7KiZMQKHyOv7Eygtd7bfiKMt0RH-v-rj0Y7uUSxdL_gwFbsYwCzhQw_5fGFjHxLvVzhkBwIDAQABo4IB6zCCAecwDgYDVR0PAQH_BAQDAgWgMIGJBggrBgEFBQcBAQR9MHswQgYIKwYBBQUHMAKGNmh0dHA6Ly9zZWN1cmUyLmFscGhhc3NsLmNvbS9jYWNlcnQvZ3NhbHBoYXNoYTJnMnIxLmNydDA1BggrBgEFBQcwAYYpaHR0cDovL29jc3AyLmdsb2JhbHNpZ24uY29tL2dzYWxwaGFzaGEyZzIwVwYDVR0gBFAwTjBCBgorBgEEAaAyAQoKMDQwMgYIKwYBBQUHAgEWJmh0dHBzOi8vd3d3Lmdsb2JhbHNpZ24uY29tL3JlcG9zaXRvcnkvMAgGBmeBDAECATAJBgNVHRMEAjAAMD4GA1UdHwQ3MDUwM6AxoC-GLWh0dHA6Ly9jcmwyLmFscGhhc3NsLmNvbS9ncy9nc2FscGhhc2hhMmcyLmNybDAxBgNVHREEKjAoghR3d3cuc2hvcmVhcmNoaWVmLmNvbYIQc2hvcmVhcmNoaWVmLmNvbTAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwHQYDVR0OBBYEFJhFov73PPyNAO6fiTaVWaoUnlltMB8GA1UdIwQYMBaAFPXN1TwIUPlqTzq3l9pWg-Zp0mj3MBMGCisGAQQB1nkCBAMBAf8EAgUAMA0GCSqGSIb3DQEBCwUAA4IBAQAeV2oaeX5sBPUUzBRoa4MpfoZ8icAdzezy_Z1DEE-GmMqAya1RFL2DCMk2rPX133YHpC3pWkDMdl6gmrz5KOD_080eUIs_VE9unD1zUPvHTA9a-Bokz_Rprqj9vq0VUuGI1Cp7xlYxs-gAv0ZTNchgsG7Fbuwz962KZAUBlzn_x0c-vXmKcz0sQJdtb2nn-vupqR0rCP0NAhKdNMaRxgOEZuZj1IAogNwBeNUVcIaGKxM4sOOwdB_DjMUvTHn0whTTr17ggAPYb33w7VN7m0uNs5RhHGQnAXdvLGOSkeqBWqYdsnNJiFvyS3cU6-Psd6MDUeKVNA0mPyYW4pbhLfAB) of a violating certificate. Most notably the `rsaEncryption` `SEQUENCE` appears with a single element as:

```
      SEQUENCE (1 elem)
        OBJECT IDENTIFIER 1.2.840.113549.1.1.1 rsaEncryption (PKCS #1)
```

When it should be a two element `SEQUENCE` containing both the OID and the `NULL` parameters:
```
      SEQUENCE (2 elem)
        OBJECT IDENTIFIER 1.2.840.113549.1.1.1 rsaEncryption (PKCS #1)
        NULL
```

**Comment by cpu** [comment]
I put a PR up for this in #285.

**Comment by cpu** [comment]
After discussion and some collaboration we've arrived at a better lint for this issue in https://github.com/zmap/zlint/pull/286 replacing my naive version in #285.

**Comment by cpu** [comment]
Resolved by https://github.com/zmap/zlint/pull/286 thanks to @tadukurow & co. :tada:
